packages/typegpu/src/data/attributes.ts

```diff
@@ -186,6 +193,68 @@ export function location<TLocation extends number, TData extends AnyData>(
   return attribute(data, { type: '@location', value: location }) as any;
 }
 
+/**
+ * Specifies how user-defined vertex shader output (fragment shader input)
+ * must be interpolated.
+ *
+ * Tip: Integer outputs cannot be interpolated.
+ *
+ * @example
+ * const Data = d.ioStruct({
+ *   a: d.f32, // has implicit 'perspective, center' interpolation
+ *   b: d.interpolate('linear, sample', d.f32),
+ * });
+ *
+ * @param location The explicit numeric location.
+ * @param data The data-type to wrap.
+ */
+export function interpolate<
+  TInterpolation extends PerspectiveOrLinearInterpolationType,
+  TData extends PerspectiveOrLinearInterpolatableData,
+>(
+  interpolationType: TInterpolation,
+  data: TData,
+): Decorate<Exotic<TData>, Interpolate<TInterpolation>>;
+
+/**
+ * Specifies how user-defined vertex shader output (fragment shader input)
+ * must be interpolated.
+ *
+ * Tip: Default sampling method of `flat` is `first`. Unless you specifically
+ * need deterministic behavior provided by `'flat, first'`, prefer explicit
+ * `'flat, either'` as it could be slightly faster in hardware.
+ *
+ * @example
+ * const Data = d.ioStruct({
+ *   a: d.f32, // has implicit 'perspective, center' interpolation
+ *   b: d.interpolate('flat, either', d.u32), // integer outputs cannot interpolate
+ * });
+ *
+ * @param location The explicit numeric location.
```

Comment:
Tiny nit, mismatched docs 📚
```suggestion
 * @param interpolationType How data should be interpolated.
```